docs/current_docs/integrations/734201-gitlab.mdx

```diff
@@ -4,9 +4,30 @@ slug: /integrations/734201/gitlab
 
 # GitLab CI
 
-The following code sample demonstrates how to integrate Dagger with GitLab CI.
+The following code samples demonstrate how to integrate Dagger with GitLab.
 
-```yaml title=".gitlab-ci.yml" file=./snippets/gitlab.yml
+## Docker executor
+
+The following code listing illustrates Dagger usage in a GitLab CI/CD pipeline, assuming either [GitLab-hosted runners](https://docs.gitlab.com/ee/ci/runners/index.html) using the (default) [Docker Machine executor](https://docs.gitlab.com/runner/executors/docker_machine.html) or [self-managed GitLab Runners](https://docs.gitlab.com/runner/install/index.html) using the [Docker executor](https://docs.gitlab.com/runner/executors/docker.html). In both these cases, the Dagger Engine is provisioned "just in time" using a Docker-in-Docker (`dind`) service.
+
+```yaml title=".gitlab-ci.yml" file=./snippets/gitlab-docker.yml
+```
+
+## Kubernetes executor
+
+The following code listing illustrates Dagger usage in a GitLab CI/CD pipeline, assuming [self-managed GitLab Runners](https://docs.gitlab.com/runner/install/index.html) in a Kubernetes cluster and using the [Kubernetes executor](https://docs.gitlab.com/runner/executors/kubernetes/).
+
+```yaml title=".gitlab-ci.yml" file=./snippets/gitlab-kubernetes.yml
 ```
 
-"Self-hosted CI runner + Kubernetes Daemonset"
+In this case, each GitLab Runner must be configured to only run on nodes with pre-provisioned instances of the Dagger Engine. This is achieved using taints and tolerations on the nodes, and pod affinity.
+
+The following code listings illustrate the configuration to be applied to each GitLab Runner, with taints, tolerations and pod affinity set via the `dagger-node` key. For an example of the corresponding node configuration, refer to the [OpenShift](./248292-openshift.mdx) integration page.
```

Comment:
I have updated the tolerations and affinity values in the OpenShift instructions to use `dagger-node`, so a user can easily match the runner configmap below with the node config. Hope that's OK @gerhard ?

I would like to also link to our Kubernetes integration page for a similar example of K8s configuration. However, the Kubernetes page uses our Helm chart with different values for tolerations and affinity: `dagger-runner` and `actions-runner`. See https://github.com/dagger/dagger/blob/main/helm/dagger/values.yaml. Does it make sense to update those values to also be `dagger-node`, so that it syncs with the configmap below?

Reply:
@gerhard for above questions

Reply:
1. Yes.
2. I like it! Good change 👍